xdsl/irdl/declarative_assembly_format.py

```diff
@@ -159,11 +163,25 @@ def parse(
             properties = state.properties
         else:
             properties = op_def.split_properties(state.attributes)
+
+        # Get the regions and cast them to the type needed in op_type
+        regions = cast(
+            Sequence[
+                Region
+                | Sequence[Operation]
+                | Sequence[Block]
+                | Sequence[Region | Sequence[Operation] | Sequence[Block]]
+                | None
+            ],
+            state.regions,
+        )
```

Comment:
I'm surprised this is needed, I'll clone and play with this locally quickly

Reply:
https://github.com/xdslproject/xdsl/commit/a7ba1d417b16f0386e4a019704ea750dcdb43715

Reply:
I don't think the `None` in the inner `list[Region | None]` is needed, might as well remove?

Reply:
Thank you, that seems sensible

Reply:
Feel free to merge those changes into this PR, then I think this is GTG

Reply:
Thanks, have done so now